raw/fetch.py

```diff
@@ -0,0 +1,224 @@
+    """Return the year, month, first date, and last date of the input month.
+
+    Parameters:
+        first_day (datetime.date): The first day of the month.
+        last_day (datetime.date): The last day of the month.
+
+    Returns:
+        tuple: A tuple containing the year, month, first date, and last date of the 
+                month as strings.
+    """
+    year, month = str(first_day)[:4], str(first_day)[5:7]
+    first_date, last_date = str(first_day)[8:], str(last_day)[8:]
+    return (year, month, first_date, last_date)
+
+
+def get_previous_month_dates() -> (Tuple[datetime.date, datetime.date, datetime.date,
+                                         datetime.date, str, str, str, str]):
+    """Return the first and third previous month's date from the current date.
+
+    Returns:
+        tuple: A tuple containing the first and third previous month's dates as 
+            datetime.date objects, and the year, month, first date, and last date 
+            of the third previous month as strings.
+    """
+
+    today = datetime.date.today()
+    # Calculate the correct previous month considering months from 1 to 12
+    prev_month = today.month + 10 if today.month < 3 else today.month - 2
+    third_prev_month = today.replace(month=prev_month)
+    first_prev_month = today.replace(month=today.month)
+    first_day_third_prev, last_day_third_prev = get_month_range(third_prev_month)
+    first_day_first_prev, last_day_first_prev = get_month_range(first_prev_month)
+
+    sl_year, sl_month, sl_first_date, sl_last_date = get_single_level_dates(
+        first_day_third_prev, last_day_third_prev
+    )
+
+    return (first_day_first_prev, last_day_first_prev, first_day_third_prev,
+        last_day_third_prev, sl_year, sl_month, sl_first_date, sl_last_date)
+
+
+def update_config_files(directory: str, field_name: str, 
+                        additional_content: str) -> None:
+    """Update the configuration files in the specified directory.
+
+    Parameters:
+        directory (str): The path to the directory containing the configuration files.
+        field_name (str): The name of the field to be updated with the new value.
+        additional_content (str): The additional content to be added under the 
+                    '[selection]' section.
+    """
+    (first_day_first_prev, last_day_first_prev, first_day_third_prev,
+    last_day_third_prev, sl_year, sl_month, sl_first_date,
+    sl_last_date) = get_previous_month_dates()
+
+    for filename in os.listdir(directory):
+        single_level_file = False
+        co_file = False
+        if filename.endswith(".cfg"):
+            if "sfc" in filename:
+                single_level_file = True
+            if "hourly" in filename:
+                co_file = True
+            config_file = os.path.join(directory, filename)
+            new_config_file(config_file, field_name, additional_content,
```

Comment:
This method should accept the dictionary arguments as kwargs.

Reply:
Code updated as per your suggestion. Thanks for the input!